### PR TITLE
Time slice cross validation MMM Notebook

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pymc-labs/pymc-marketing/issues/956

----
📚 Documentation preview 📚: https://pymc-marketing--971.org.readthedocs.build/en/971/

### CODE REVIEW

File: pymc_marketing/metrics.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    The CRPS — Continuous Ranked Probability Score — is a score function that compares a
+    single ground truth value to a Cumulative Distribution Function.
+
+    Parameters
+    ----------
+    y_true : array-like
+        The ground truth values.
+    y_pred : array-like
+        The predicted values. It is expected that y_pred has one extra sample
+        dimension on the left.
```
Comment (wd60622) [inline pymc_marketing/metrics.py:32]: Can you add an example of what these dims look like in the docstring. For instance, (date, ) and (sample, date) or (model, date) and (sample, model, date) 

Also that the leftmost dim is likely the samples? Is that the case for our models?

File: pymc_marketing/metrics.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Evaluation Metrics."""
+
+import numpy as np
+import numpy.typing as npt
+
+
+def per_observation_crps(y_true: npt.NDArray, y_pred: npt.NDArray) -> npt.NDArray:
```
Comment (wd60622) [inline pymc_marketing/metrics.py:20]: Is this per_observation or per_sample? Doesn't this get a distribution of the metric?

Maybe `calculate_crps_distribution` or something to hint that this is a distribution of a metric

File: pymc_marketing/metrics.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    y_true : array-like
+        The ground truth values.
+    y_pred : array-like
+        The predicted values. It is expected that y_pred has one extra sample
+        dimension on the left.
+
+    Returns
+    -------
+    array-like
+        The CRPS for each observation.
```
Comment (wd60622) [inline pymc_marketing/metrics.py:37]: Might be having this wrong in my previous comments. Could you provide an explicit example of the samples here. f((date, ), (sample, date)) -> (date, )? or (sample, )? Which dim is being reduced?

**Review by wd60622 — COMMENTED** [review]
Overall looks really good. Some questions. Looking more into the notebook now

**Comment by juanitorduz** [comment]
<p>I think usually people woill look into the average (crps)</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/971/discussion/#comment-2316650125'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by juanitorduz** [comment]
<p>Thanks!</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/971/discussion/#comment-2316650143'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: pymc_marketing/metrics.py (hunk 2 above)
Comment (juanitorduz) [reply pymc_marketing/metrics.py:20]: added an example in https://github.com/pymc-labs/pymc-marketing/pull/971/commits/d70e77ec88d5b85306fc7d9d419644c8cad55b1a

File: pymc_marketing/metrics.py (hunk 3 above)
Comment (juanitorduz) [reply pymc_marketing/metrics.py:37]: Added an example in https://github.com/pymc-labs/pymc-marketing/pull/971/commits/d70e77ec88d5b85306fc7d9d419644c8cad55b1a

File: pymc_marketing/metrics.py (hunk 1 above)
Comment (juanitorduz) [reply pymc_marketing/metrics.py:32]: I added an example in https://github.com/pymc-labs/pymc-marketing/pull/971/commits/d70e77ec88d5b85306fc7d9d419644c8cad55b1a . Hopefully it clarifies (It was confusing for me)

**Review by wd60622 — APPROVED** [review]
Looks great. Thanks for the adjustments

**Pull request merged**